data/secscan_model/secscan_v4_model.py

```diff
@@ -202,11 +202,23 @@ def needs_reindexing_query(indexer_hash):
 
         # TODO(alecmerdler): We want to index newer manifests first, while backfilling older manifests...
         iterator = itertools.chain(
+            # Guarantees at least one batch of most recent manifests
             yield_random_entries(
                 not_indexed_query,
                 Manifest.id,
                 batch_size,
                 max_id,
+                # Because we have multiple workers and must stay random to ensure minimal duplicative work
+                # we will instead limit the set worked to the most recent batch.
+                max_id - batch_size,
+            ),
```

Comment:
Won't this stop doing any work once `(max_id - batch_size) ... max_id` is done?

Reply:
Yes but `max_id` keeps getting updated every time the worker does a `perform_indexing` call

Reply:
Isn't it the `min_id` that gets updated? 
i.e the `max_id` if the last batch becomes the new `min_id`

Reply:
both get updated
```
        min_id = (
            start_token.min_id
            if start_token is not None
            else Manifest.select(fn.Min(Manifest.id)).scalar()
        )
        max_id = Manifest.select(fn.Max(Manifest.id)).scalar()
```